pkg/cluster/px/px.go

```diff
@@ -243,3 +256,217 @@ func (ops *pxClusterOps) runDockerPuller(imageToPull string) error {
 
 	return nil
 }
+
+func (ops *pxClusterOps) scaleSharedAppsToZero() error {
+	sharedDeps, sharedSS, err := ops.getPXSharedApps()
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("found %d deployments and %d statefulsets to scale down", len(sharedDeps), len(sharedSS))
+
+	var valZero int32
+	for _, d := range sharedDeps {
+		deploymentName := d.Name
+		deploymentNamespace := d.Namespace
+		logrus.Infof("scaling down deployment: [%s] %s", deploymentNamespace, deploymentName)
+
+		t := func() (interface{}, bool, error) {
+			dCopy, err := ops.kubeClient.Apps().Deployments(deploymentNamespace).Get(deploymentName, metav1.GetOptions{})
+			if err != nil {
+				if matched, _ := regexp.MatchString(".+ not found", err.Error()); matched {
+					return nil, false, nil // done as deployment is deleted
+				}
+
+				return nil, true, err
+			}
+
+			dCopy = dCopy.DeepCopy()
+			dCopy.Annotations[replicaMemoryKey] = fmt.Sprintf("%d", *dCopy.Spec.Replicas)
```

Comment:
May want to add a comment to as what this is doing and why it is needed

Reply:
done